xc/device.ml

```diff
@@ -1748,18 +1748,32 @@ module Vusb = struct
     && not (List.mem driver_id (qom_list ~xs ~domid))then
       qmp_send_cmd domid Qmp.(Device_add Device.({driver; device=USB { USB.id=driver_id; params=None }})) |> ignore
 
-  let vusb_plug ~xs ~privileged ~domid ~id ~hostbus ~hostport ~version =
+  let vusb_plug ~xs ~privileged ~domid ~id ~hostbus ~hostport ~version ~speed =
     debug "vusb_plug: plug VUSB device %s" id;
-    let get_bus v =
-      if String.startswith "1" v then
-        "usb-bus.0"
-      else begin
-        (* Here plug usb controller according to the usb version*)
-        let usb_controller_driver = "usb-ehci" in
-        let driver_id = "ehci" in
-        vusb_controller_plug ~xs ~domid ~driver:usb_controller_driver ~driver_id;
-        Printf.sprintf "%s.0" driver_id;
-      end
+    let usb_bus0 = "usb-bus.0" in
+    let ehci0    = "ehci.0" in
+    let get_bus () =
+      let bus_from_speed =
+        match float_of_string_opt speed with
+        | None -> None
+        | Some speed -> if 0. < speed && speed <= 480. then
```

Comment:
If we envision support for more usb devices, we could turn this into a variant
```
let speed_from_float = function
| 1.5 -> `Low (* 1.0 *)
| 12. -> `Full (* 1.1 *)
| 480. -> `Hi (* 2.0 *)
| 5000. -> `Super (* 3.0 *)
| f when f >= 10000. -> `SuperPlus (* fold all future speeds into 3.1 ?*)
| _ -> Erroneous speed reported
```

Reply:
I think matching floats exactly is not a good idea. I think this needs a sequence of tests using `<=`.

Reply:
Float equality may be an issue, it's strange we can only get the speed class as a float: it's defined as an [enum](https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/usb/ch9.h#L1137) in Linux